Arm/Arm64 cross builds failing

As of:

http://dotnet-ci.cloudapp.net/job/dotnet_coreclr_arm64_linux_debug/29/

Changes in that build:

Changes 1.Add link to .NET Core Roadmap (commit: 2c655822e3e26f9f3edea50ac8f4d4c91873a5bf) (detail)
2.Use Frame list context when GetThreadContext is unavailable (commit: eb3260e67bc653b56a80fbf0e8a07d78e3a404c1) (detail)
3.Use out-of-proc libunwind to unwind native stack from DAC (commit: d0f71d0b22e9012263e4a078d24738c75e384a90) (detail)

Error:

In file included from /mnt/j/workspace/dotnet_coreclr_arm64_linux_debug/src/pal/src/exception/seh.cpp:282:
/mnt/j/workspace/dotnet_coreclr_arm64_linux_debug/src/pal/src/exception/seh-unwind.cpp:494:5: error: use of undeclared identifier 'WinContextToUnwindContext'; did you mean 'WinContextToUnwindCursor'?
    WinContextToUnwindContext(context, &unwContext);
    ^~~~~~~~~~~~~~~~~~~~~~~~~
    WinContextToUnwindCursor
/mnt/j/workspace/dotnet_coreclr_arm64_linux_debug/src/pal/src/exception/seh-unwind.cpp:88:13: note: 'WinContextToUnwindCursor' declared here
static void WinContextToUnwindCursor(CONTEXT *winContext, unw_cursor_t *cursor)
            ^
/mnt/j/workspace/dotnet_coreclr_arm64_linux_debug/src/pal/src/exception/seh-unwind.cpp:494:40: error: cannot initialize a parameter of type 'unw_cursor_t *' (aka 'unw_cursor *') with an rvalue of type 'unw_context_t *' (aka 'unw_tdep_context *')
    WinContextToUnwindContext(context, &unwContext);
                                       ^~~~~~~~~~~
/mnt/j/workspace/dotnet_coreclr_arm64_linux_debug/src/pal/src/exception/seh-unwind.cpp:88:73: note: passing argument to parameter 'cursor' here
static void WinContextToUnwindCursor(CONTEXT *winContext, unw_cursor_t *cursor)

Different error on arm64:

In file included from /mnt/j/workspace/dotnet_coreclr_arm64_linux_debug/src/pal/src/exception/seh.cpp:282:
/mnt/j/workspace/dotnet_coreclr_arm64_linux_debug/src/pal/src/exception/seh-unwind.cpp:398:14: error: duplicate case value: 'UNW_REG_IP' and 'UNW_AARCH64_X30' both equal '30'
        case UNW_AARCH64_X30:  *valp = (unw_word_t) winContext->Lr; break;
             ^
/mnt/j/workspace/dotnet_coreclr_arm64_linux_debug/src/pal/src/exception/seh-unwind.cpp:395:14: note: previous case defined here
        case UNW_REG_IP:       *valp = (unw_word_t) winContext->Pc; break;
             ^
1 error generated.

--- a/src/debug/inc/dbgipcevents.h
+++ b/src/debug/inc/dbgipcevents.h
@@ -181,7 +181,7 @@ struct MSLAYOUT DebuggerIPCRuntimeOffsets
 // aren't any embedded buffers in the DebuggerIPCControlBlock).
 
 #if defined(DBG_TARGET_X86) || defined(DBG_TARGET_ARM)
-#define CorDBIPC_BUFFER_SIZE (1500) // hand tuned to ensure that ipc block in IPCHeader.h fits in 1 page.
+#define CorDBIPC_BUFFER_SIZE (2088) // hand tuned to ensure that ipc block in IPCHeader.h fits in 1 page.
 #else  // !_TARGET_X86_ && !_TARGET_ARM_
 // This is the size of a DebuggerIPCEvent.  You will hit an assert in Cordb::Initialize() (DI\process.cpp)
 // if this is not defined correctly.  AMD64 actually has a page size of 0x1000, not 0x2000.

--- a/src/pal/src/exception/seh-unwind.cpp
+++ b/src/pal/src/exception/seh-unwind.cpp
@@ -323,7 +323,8 @@ BOOL PAL_VirtualUnwind(CONTEXT *context, KNONVOLATILE_CONTEXT_POINTERS *contextP
 #error don't know how to unwind on this platform
 #endif
 
-
+// These methods are only used on the AMD64 build
+#ifdef _AMD64_
 #ifdef __LINUX__
 
 
@@ -539,6 +540,7 @@ BOOL PAL_VirtualUnwindOutOfProc(CONTEXT *context,
 }
 
 #endif // !__LINUX__
+#endif // _AMD64_
 
 /*++
 Function:

